src/preamble.js

```diff
@@ -923,6 +952,10 @@ function createWasm() {
 #endif
 #if WASM_BACKEND && ASYNCIFY
     exports = Asyncify.instrumentWasmExports(exports);
+#endif
+#if ASSERTIONS
+    Module['__orig_asm'] = exports;
+    exports = wrapNativeFunctionsWithAssertions(exports);
```

Comment:
could the name of this and the asyncify method above it be more in harmony? perhaps `instrumentWasmExportsWithAssertions`.